### PR TITLE
Enhance command push classify-eval

### PR DESCRIPTION
**Pull request by La0**
- Add more logging:
   - around errors
   - on recalculate normal flow
   - on Taskcluster task fetch normal flow
- Refactor the output & inner triage of push classification so it can generate both:
   - same summary output
   - a CSV output with more detailed informations (tracking exceptions)

### CODE REVIEW

File: mozci/console/commands/push.py [hunk 1]
```diff
@@ -232,23 +235,23 @@ def handle(self):
...
                 try:
                     classification, _ = push.classify(
                         confidence_medium=medium_conf, confidence_high=high_conf
                     )
-                except Exception:
-                    classification_failed.append(push)
-                    continue
+                    self.line(
+                        f"<comment>Classified {branch} {push.rev} : {classification.name}</comment>"
+                    )
```
Comment (marco-c) [inline mozci/console/commands/push.py:249]: Might be too noisy if we run an analysis on a month worth of pushes

File: mozci/console/commands/push.py [hunk 2]
```diff
@@ -260,48 +263,94 @@ def handle(self):
...
-                if push.backedout:
-                    unknown_backedout_pushes.append(push)
-                else:
-                    unknown_non_backedout_pushes.append(push)
+                    self.classifications[push] = classification = PushStatus[
+                        artifact["push"]["classification"]
+                    ]
+                    self.line(
+                        f"<comment>Found classification {branch} {push.rev} : {classification.name}</comment>"
+                    )
```
Comment (marco-c) [inline mozci/console/commands/push.py:271]: Same here

File: mozci/console/commands/push.py [hunk 3]
```diff
@@ -269,48 +276,96 @@ def handle(self):
...
+        self.log_pushes(PushStatus.UNKNOWN, True)
+
+        output = self.option("output")
+        if output:
+            # Build stats for CSV
+            with open(output, "w") as csvfile:
+                writer = csv.DictWriter(
+                    csvfile,
+                    fieldnames=[
+                        "push",
```
Comment (marco-c) [inline mozci/console/commands/push.py:319]: We could remove this as the `revision` in a branch already uniquely identifies a push.

**Review by marco-c — APPROVED** [review]
LGTM

File: mozci/console/commands/push.py (hunk 3 above)
Comment (La0) [reply mozci/console/commands/push.py]: Fixed in https://github.com/mozilla/mozci/pull/600/commits/e6d66b06884df15cc6bbd8a48b216b108c9af184

**Pull request merged**